ysh/expr_to_ast.py

```diff
@@ -691,7 +691,11 @@ def Expr(self, pnode):
                 return cast(BracedVarSub, pnode.GetChild(1).tok)
 
             elif typ == grammar_nt.dq_string:
-                return cast(DoubleQuoted, pnode.GetChild(1).tok)
+                s = cast(DoubleQuoted, pnode.GetChild(1).tok)
+                # sugar: ^"..." is short for ^["..."]
+                if pnode.GetChild(0).tok.id == Id.Left_CaretDoubleQuote:
+                    return expr.Literal(s)
```

Comment:
It might also be possible to change the evaluator, i.e. `ysh/expr_eval.py`

If a `DoubleQuoted` node has  an `Id.Left_CaretDoubleQuote`, then it could "stop" evaluating it

It may depend if there are some "LST invariants" we would to preserve, but we can think about that later I guess

This has a good test, which helps any future refactoring!